utils/network_linux.go

```diff
@@ -416,3 +417,47 @@ func writeProcSys(path, value string) error {
 	}
 	return err
 }
+
+// CleanUpNamespace deletes the devices in the network namespace.
+func CleanUpNamespace(args *skel.CmdArgs, logger *logrus.Entry) error {
+	// Only try to delete the device if a namespace was passed in.
+	if args.Netns != "" {
+		logger.WithFields(logrus.Fields{
+			"netns": args.Netns,
+			"iface": args.IfName,
+		}).Debug("Checking namespace & device exist.")
+		devErr := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+			_, err := netlink.LinkByName(args.IfName)
+			return err
+		})
+
+		if devErr == nil {
+			fmt.Fprintf(os.Stderr, "Calico CNI deleting device in netns %s\n", args.Netns)
+			// Deleting the veth has been seen to hang on some kernel version. Timeout the command if it takes too long.
+			ch := make(chan error, 1)
+
+			go func() {
+				err := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+					return ip.DelLinkByName(args.IfName)
```

Comment:
Note to reviewer: this line changed vs the version in utils.go that I removed.

Old line was `_, err := ip.DelLinkByNameAddr(args.IfName, netlink.FAMILY_V4)` but that method changed signature so I had to fix up and it looked like `ip.DelLinkByName()` was a better fit for what we wanted here anyway?

Reply:
Seems reasonable.